### PR TITLE
Trap reverts in poster, uppercase symbols, and 6 decimals w/ some minor other fixes

### PR DESCRIPTION
**Pull request by jflatow**

### CODE REVIEW

File: sdk/javascript/src/reporter.ts [hunk 1]
```diff
@@ -40,7 +45,12 @@ export function fancyParameterEncoder(paramType: string): [string, (any) => any]
   // Decimals are encoded as uints with 18 decimals of precision on-chain.
   if (paramType === 'decimal') {
     actualParamType = 'uint256';
-    actualParamEnc = (x) => web3.utils.toBN('1000000000000000000').muln(x).toString();
+    actualParamEnc = (x) => web3.utils.toBN(1e6).muln(x).toString();
+  }
+
+  if (paramType == 'symbol') {
+    actualParamType = 'string';
+    actualParamEnc = (x) => x.toUpperCase();
```
Comment (hayesgm) [inline sdk/javascript/src/reporter.ts]: upppppppppercase??????

Comment (jflatow) [reply sdk/javascript/src/reporter.ts]: You mean which do we choose? Rob suggested upper

Comment (hayesgm) [reply sdk/javascript/src/reporter.ts]: yeah just wondering why since upper case letter are so DiFfiCulT to write.

Comment (jflatow) [reply sdk/javascript/src/reporter.ts]: I guess it makes sense since tickers are usually upper 🤷‍♂

Comment (hayesgm) [reply sdk/javascript/src/reporter.ts]: hm, okay, do it!

**Pull request merged**